apps/api/src/app/widgets/usecases/remove-message/remove-message.usecase.ts

```diff
@@ -0,0 +1,95 @@
+import { Inject, Injectable, NotFoundException } from '@nestjs/common';
+import {
+  MessageEntity,
+  DalException,
+  MessageRepository,
+  SubscriberRepository,
+  SubscriberEntity,
+  MemberRepository,
+} from '@novu/dal';
+import { ChannelTypeEnum } from '@novu/shared';
+import { AnalyticsService } from '@novu/application-generic';
+
+import { CacheKeyPrefixEnum } from '../../../shared/services/cache';
+import { QueueService } from '../../../shared/services/queue';
+import { ANALYTICS_SERVICE } from '../../../shared/shared.module';
+import { RemoveMessageCommand } from './remove-message.command';
+import { InvalidateCache } from '../../../shared/interceptors';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { MarkEnum } from '../mark-message-as/mark-message-as.command';
+
+@Injectable()
+export class RemoveMessage {
+  constructor(
+    private messageRepository: MessageRepository,
+    private queueService: QueueService,
+    @Inject(ANALYTICS_SERVICE) private analyticsService: AnalyticsService,
+    private subscriberRepository: SubscriberRepository,
+    private memberRepository: MemberRepository
+  ) {}
+
+  @InvalidateCache([CacheKeyPrefixEnum.MESSAGE_COUNT, CacheKeyPrefixEnum.FEED])
+  async execute(command: RemoveMessageCommand): Promise<MessageEntity> {
+    const subscriber = await this.subscriberRepository.findBySubscriberId(command.environmentId, command.subscriberId);
+    if (!subscriber) throw new NotFoundException(`Subscriber ${command.subscriberId} not found`);
+
+    let deletedMessage;
+    try {
+      await this.messageRepository.delete({
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+        _id: command.messageId,
```

Comment:
Could you please add here also the _subscriberId? To make sure that subscribers can't delete other peoples messages :)